examples/hello-recursion.yaml

```diff
@@ -0,0 +1,43 @@
+# The following workflow executes a workflow which the number of next level depends on the output of the previous step.
```

Comment:
```suggestion
# This example shows how to use recursion in a Workflow.
# The number of executions of the next step depends on the output of the previous step.
```

- "the following workflow executes a workflow" is redundant, be more simple and direct [per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- "level" is not a term in Workflows and "number of next level" is missing a concrete subject -- let's use the word "executions" here